### DISCUSSION TITLE
Feature Suggestion: Sega CD Transfer Suite Integration

### DISCUSSION THREAD

**Discussion by ca7bc0c5f2** [retronas/discussions#95] [Ideas]
It would be awesome if RetroNAS could support direct Megadrive/32X dumping using a custom parallel cable and the Sega CD Transfer Suite. https://www.retrodev.com/transfer.html. This is all win32 code, and doesn't like to run on 64 bit versions of Windows. I'm not the original writer, just a user. I'd love to see this could be rewritten, or made working in Wine then integrated into the menu system.

**Comment by sairuk** [comment]
I am a fan of Mike's tools, I have this adapter built and a modded console to use with it. I normally apply this to dump my own carts, I haven't yet used the transfer suite side of things, its been on my list of things to do for years. I can test at least test it.

Can you drop some some basic instructions here with the specific usage of transfer.exe with an example in this issue?

In regards to RetroNAS integration this would fall under _on device management_ which we need to think about what is in scope

In this space I already have a note to investigate integrating `ucon64` as a general purpose tool for many devices but I need to see if i can get it to compile on arm first, then spec that out across platforms

I don't know how I feel about wine overhead for windows tools at this stage, something else to think about

**Comment by ca7bc0c5f2** [comment]
My apologies for any confusion. I too use the adapter and tools to dump my own carts. I have not actually utilized the transfer function.

Big thanks to mentioning ucon64 and considering it. I wasn't even aware this tool existed. Given it supports the Sega CD transfer suite, it may fulfill the intent of this feature request. 

I'm running RetroNAS on x86. I'll do some experimenting.

**Comment by sairuk** [comment]
ucon64 on rpi can't be compiled with parallel device support on due to a missing libc header, arguably fair enough no parallel ports on the thing. I will have to muck around and see what I can do before this becomes a reality, ucon64 on rpi has been a thing i've wanted for a while so am happy to give it a go.

**Comment by sairuk** [comment]
I have now added uCON64 support to RetroNAS, need to do some more testing when I get a chance but its compiled and available. It is installed from the On-Device Management menu
